x/rollapp/keeper/block_height_to_finalization_queue.go

```diff
@@ -141,26 +146,99 @@ func (k *Keeper) finalizePendingState(ctx sdk.Context, stateInfoIndex types.Stat
 	return nil
 }
 
-func (k Keeper) updateQueueForHeight(ctx sdk.Context, blockHeightToFinalizationQueue types.BlockHeightToFinalizationQueue, failedRollapps map[string]uint64) {
-	// remove the blockHeightToFinalizationQueue if all the rollapps' states are finalized
-	if len(failedRollapps) == 0 {
-		k.RemoveBlockHeightToFinalizationQueue(ctx, blockHeightToFinalizationQueue.CreationHeight)
+func (k Keeper) updateQueueForHeight(ctx sdk.Context, queue types.BlockHeightToFinalizationQueue, failedRollapps map[string]uint64) {
+	// remove the queue if all the rollapp states are finalized
```

Comment:
I think this logic needs to be further reworked, in particular updating the queue from the `failedRollapps` map, because previously we had a signle queue with state updates from mixed rollapps, but now it's grouped, so `updateQueueForHeight` will do so for the current queue's rollapp. 

Also, `finalizeQueueForHeight` should be changed to simply stop finalizing at the first failure for the current queue/rollapp.

Reply:
thanks for the catch, reworked `FinalizeAllPending`